packages/algob/src/types.ts

```diff
@@ -385,14 +385,37 @@ export interface Checkpoint {
 export type ASADef = z.infer<typeof ASADefSchema>;
 export type ASADefs = z.infer<typeof ASADefsSchema>;
 
-export interface GrpTxnParams {
-  fromAccount: Account
-  toAccountAddr: string
-  lsig: LogicSig
+export type execParams = AlgoTransferParam | AssetTransferParam | SSCCallsParam;
+
+export interface Sign {
+  sign: "sk" | "lsig"
+  lsig?: LogicSig
+}
+
+export interface AlgoTransferParam extends Sign {
+  type: "algo"
+  fromAccount: AccountSDK
+  toAccountAddr: AccountAddress
   amountMicroAlgos: number
   payFlags: TxParams
 }
 
+export interface AssetTransferParam extends Sign {
+  type: "asset"
+  fromAccount: AccountSDK
+  toAccountAddr: AccountAddress
+  amount: number
+  assetID: number
+  payFlags: TxParams
+}
+
+export interface SSCCallsParam extends SSCOptionalFlags, Sign {
+  type: "clearSSC" | "deleteSSC" | "callNoOpSSC" | "closeSSC"
```

Comment:
This should be ENUM and not a string. In general `execParams.type` should be an enum. As we shouldn't ask the user to pass strings as txn type. Eg - user can pass 'clearssc' instead of 'clearSSC'.

Reply:
yes, thanks